support/export/rmtab.c

```diff
@@ -96,3 +97,74 @@ rmtab_read(void)
 	}
 	return 0;
 }
+
+/*
+ * The rmtab file can acculmulate stale entries over time. So when the list of
+ * exports changes, make a best effort attempt to insure that every entry in
+ * rmtab has a corresonding export.
+ */
+void
+rmtab_rebuild(void)
+{
+	struct rmtabent	*rep;
+	struct stat	stb;
+	FILE		*fp;
+	int		rmtab_lockid, etab_lockid;
+	int		entries_seen = 0, entries_copied = 0;
+
+	/* Hold locks for the rmtab and etab files during rebuilding */
+	if ((rmtab_lockid = xflock(_PATH_RMTABLCK, "w")) < 0)
+		return;
+
+	/* fast path for no NFSv3 mounts */
+	if (stat(_PATH_RMTAB, &stb) < 0 || stb.st_size == 0) {
+		xfunlock(rmtab_lockid);
+		return;
+	}
+
+	if ((etab_lockid = xflock(_PATH_ETABLCK, "r")) < 0) {
+		xfunlock(rmtab_lockid);
+		return;
+	}
+
+	if (!setrmtabent("r")) {
+		xfunlock(rmtab_lockid);
+		xfunlock(etab_lockid);
+		return;
+	}
+	if (!(fp = fsetrmtabent(_PATH_RMTABTMP, "w"))) {
+		endrmtabent();
+		xfunlock(rmtab_lockid);
+		xfunlock(etab_lockid);
+		return;
+	}
+
+	/* Visit all the entries in rmtab file */
+	while ((rep = getrmtabent(1, NULL)) != NULL) {
+		struct exportent *xp;
+
+		/* filter entries that are not exported */
+		setexportent(_PATH_ETAB, "r");
+		while ((xp = getexportent_nofollow(0, 0)) != NULL) {
+			if (!strcmp(rep->r_path, xp->e_path)) {
```

Comment:
I believe that you can mount directories that are contained inside of an export. In other words, if I'm exporting `/some/path`, then an NFS client can mount `/some/path/inside/here`. I also think that some of our appdata plugins do just that (they mount the parent directory of the timeflow, but also mount child directories in other locations).

Reply:
good observation! Need to figure out how to account for those.

Reply:
I posted a new commit that also checks for nested mounts.